packages/now-node-bridge/src/bridge.ts

```diff
@@ -1,72 +1,82 @@
-const http = require('http');
+import { AddressInfo } from 'net';
+import { Server, request } from 'http';
 
 function normalizeEvent(event) {
+  let method: string;
+  let path: string;
+  let body: Buffer | string;
+  let encoding: string;
+  let headers;
   let isApiGateway = true;
 
   if (event.Action === 'Invoke') {
     isApiGateway = false;
-    const invokeEvent = JSON.parse(event.body);
-
-    const {
-      method, path, headers, encoding,
-    } = invokeEvent;
-
-    let { body } = invokeEvent;
+    ({ method, path, headers, encoding, body } = JSON.parse(event.body));
 
     if (body) {
       if (encoding === 'base64') {
-        body = Buffer.from(body, encoding);
+        body = Buffer.from(body as string, encoding);
       } else if (encoding === undefined) {
-        body = Buffer.from(body);
+        body = Buffer.from(body as string);
       } else {
         throw new Error(`Unsupported encoding: ${encoding}`);
       }
     }
-
-    return {
-      isApiGateway, method, path, headers, body,
-    };
+  } else {
+    ({ httpMethod: method, path, headers, body } = event);
   }
 
-  const {
-    httpMethod: method, path, headers, body,
-  } = event;
-
-  return {
-    isApiGateway, method, path, headers, body,
-  };
+  return { isApiGateway, method, path, headers, body };
 }
 
-class Bridge {
-  constructor() {
+export class Bridge {
+  private server: Server;
+  private listening: Promise<AddressInfo>;
+  private resolveListening: (info: AddressInfo) => void;
+
+  constructor(server?: Server) {
+    this.server = null;
+    if (server) {
+      this.setServer(server);
+    }
     this.launcher = this.launcher.bind(this);
   }
 
-  launcher(event) {
-    // eslint-disable-next-line consistent-return
-    return new Promise((resolve, reject) => {
-      if (this.userError) {
-        console.error('Error while initializing entrypoint:', this.userError);
-        return resolve({ statusCode: 500, body: '' });
-      }
+  setServer(server: Server) {
+    this.server = server;
+    this.listening = new Promise(resolve => {
+      this.resolveListening = resolve;
+    });
+    server.once('listening', () => {
+      this.resolveListening(server.address() as AddressInfo);
```

Comment:
I don't think this is safe to cast.

I checked the type and its `address(): AddressInfo | string` so maybe a better solution is the following:

```suggestion
      const addr = server.address();
      if (typeof addr === 'string') {
        throw new Error('Unexpected string for server.address() ' + addr);
      } else {
        this.resolveListening(addr);
      }
```

Reply:
That's fine too, but the `string` case only is returned when `listen()` is called with a UNIX file path, so that type will never actually occur. Erroring with an explicit message seems like a good idea though 👍